conformance/lang/expressions/string-template-expr/string_template_expr.balt

```diff
@@ -0,0 +1,323 @@
+Labels: string, function-defn, int, string-template-expr, string-literal
+
+function errorFunction() {
+    string a1 = "a1";
+    int _ = string `${a1}`; // @error expected 'int' but found 'string'
+    int a2 = 12;
+    int _ = string `${a2}`; // @error expected 'int' but found 'string'
+    int a3 = 13;
+    int _ = string `${a2 + a3}`; // @error expected 'int' but found 'string'
+    int _ = <int> string `${a2 + a3}`; // @error 'string' cannot be cast to 'int'
+}
+
+Test-Case: output
+Description: Test type of the expression in each interpolation of string-template-expr.
+Labels: string, function-defn, DecimalNumber, string-template-expr, float, int, boolean
+
+function init() {
+    string a1 = "a1";
+    string temp1 = string `${a1}`;
+    io:println(temp1); // @output a1
+    int a2 = 12;
+    string temp2 = string `${a2}`;
+    io:println(temp2); // @output 12
+    decimal a3 = 23.5d;
+    string temp3 = string `${a3}`;
+    io:println(temp3); // @output 23.5
+    decimal a4 = 23.5;
+    string temp4 = string `${a4}`;
+    io:println(temp4); // @output 23.5
+    float a5 = 23.5;
+    string temp5 = string `${a5}`;
+    io:println(temp5); // @output 23.5
+    float a6 = 0x32.5;
+    string temp6 = string `${a6}`;
+    io:println(temp6); // @output 50.3125
+    boolean a7 = true;
+    string temp7 = string `${a7}`;
+    io:println(temp7); // @output true
+    string:Char a8 = "C";
+    string temp8 = string `${a8}`;
+    io:println(temp8); // @output C
+}
+
+Test-Case: output
+Description: Test type of the expression in each interpolation of string-template-expr.
+Labels: string, function-defn, string-template-expr, int, union-type, float, decimal
+
+function init() {
+    string|int a1 = "a1";
+    string temp1 = string `${a1}`;
+    io:println(temp1); // @output a1
+    int|float|decimal a2 = 12;
+    string temp2 = string `${a2}`;
+    io:println(temp2); // @output 12
+    string|int|decimal|float|boolean a3 = true;
+    string temp3 = string `${a3}`;
+    io:println(temp3); // @output true
+    int|string:Char a4 = "C";
+    string temp4 = string `${a4}`;
+    io:println(temp4); // @output C
+    int|float|decimal|byte a5 = 12;
+    string temp5 = string `${a5}`;
+    io:println(temp5); // @output 12
+}
+
+Test-Case: output
+Description: Test type of the expression in each interpolation of string-template-expr.
+Labels: string, function-defn, string-template-expr, int, float, decimal, union-type, intersection-type
+
+function init() {
+    string|int & readonly a1 = "a1";
+    string temp1 = string `${a1}`;
+    io:println(temp1); // @output a1
+    int|float|decimal & readonly a2 = 12;
+    string temp2 = string `${a2}`;
+    io:println(temp2); // @output 12
+    string|int|decimal|float|boolean & readonly a3 = true;
+    string temp3 = string `${a3}`;
+    io:println(temp3); // @output true
+}
+
+Test-Case: output
+Description: Test type of the expression in each interpolation of string-template-expr.
```

Comment:
Please improve the descriptions for each test mentioning what exactly is being tested. There are seven tests with exactly this description atm.